src/test/scala/chiselTests/simulator/SimulatorSpec.scala

```diff
@@ -0,0 +1,84 @@
+package chiselTests.simulator
+
+import chisel3._
+import chisel3.simulator._
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.must.Matchers
+import svsim._
+
+class GCD extends Module {
+  val io = IO(new Bundle {
+    val a = Input(UInt(32.W))
+    val b = Input(UInt(32.W))
+    val loadValues = Input(Bool())
+    val result = Output(UInt(32.W))
+    val resultIsValid = Output(Bool())
+  })
+  val x = Reg(UInt(32.W))
+  val y = Reg(UInt(32.W))
+  when(x > y) { x := x -% y }.otherwise { y := y -% x }
+  when(io.loadValues) { x := io.a; y := io.b }
+  io.result := x
+  io.resultIsValid := y === 0.U
+}
+
+class VerilatorSimulator(val workspacePath: String) extends SingleBackendSimulator[verilator.Backend] {
+  val backend = verilator.Backend.initializeFromProcessEnvironment()
+  val tag = "verilator"
+  val commonCompilationSettings = CommonCompilationSettings()
+  val backendSpecificCompilationSettings = verilator.Backend.CompilationSettings()
+}
```

Comment:
I see you're extending `SingleBackendSimulator` in the tests (effectively "user code"). Do we want users to be able to implement their own Simulators?

Reply:
The "Simulator" is effectively a backend + compilation settings, so yes we expect users would extend this type if they want a specific set of compile options.

That said, I intend to provide a higher level API which will provide a "default" simulator so users can just say `simulate()` and it will do something like use Verilator with default options.